core.coffee

```diff
@@ -407,6 +408,20 @@
 			delete: (params) ->
 				return @request(params, method: 'DELETE')
 
+			batch: (data) ->
```

Comment:
This doesn't run the individual request parts through the same syntactic sugar that the rest of this module uses, so right now you'd have to manually write the odata filters for any batch requests - you should be able to do it by running the individual parts via `compile` to get the url

Reply:
I think I am doing everything. `compile` is called inside the `compileBatch`. The part I am less sure about is setting the apiRoots correctly for the requests in the batch, since it does not happen in `compile` but in the `request` function I had to duplicate some code and as it is right now it does not look 100% solid to me (although it passes all the test-cases I could think of). https://github.com/resin-io/pinejs-client-js/pull/34/files#diff-7211713139c36b3a80c048e901477db2R478